### ISSUE TITLE
Engine support in Home Assistant

### ISSUE THREAD

**Issue by TENFIRE** [molobrakos/volvooncall#23]
Hi, would you be able to update home assistant to volvooncall:0.5.0 and add a start/stop engine switch as well?

This is the feature I use the most frequently in the VOC-app, it would really be great if we had this in home assistant as well.

**Comment by molobrakos** [comment]
Yep, that's the plan! I have been waiting for the API to stabilize. Also, I want to get rid of as much voc-specific stuff as possible in HA, such as [enumerating all available resources](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/volvooncall.py#L39) and instead letting the voc lib provide all available instruments. Glad to hear the engine start functionality is working for you, it is not supported by my car so I haven't been able to test it myself at all.

**Comment by molobrakos** [comment]
Maybe you can test https://github.com/molobrakos/home-assistant/tree/new-voc and report if it work for you? (My car does not support starting/stopping engine from the api)
Diff here: https://github.com/home-assistant/home-assistant/compare/dev...molobrakos:new-voc

**Comment by molobrakos** [comment]
Oh, ok, that should be fixed now. Can you please test again?

**Comment by TENFIRE** [comment]
Heater and Engine started and stopped using the switches. Although the state for the engine was shown as off all the time, even though it actually was on. The heater updated the state as it should.

**Comment by molobrakos** [comment]
> Although the state for the engine was shown as off all the time, even though it actually was on.

By this you mean the engine status binary sensor or the engine switch? (both are available, since not all vehicles supports starting the engine, only viewing the current state)

**Comment by TENFIRE** [comment]
I mean the engine switch in homeassistant.

**Comment by molobrakos** [comment]
Added a forced update from the server after calling start/stop engine as well as lock/unlock. Maybe it works better now?

**Comment by TENFIRE** [comment]
The engine still starts fine but both the binary sensor and the switch still has the state "off" after I turned it on using the switch. I even tried restarting home assistant while the engine was running, still "off".

I tried playing around with voc and found that 'engineRunning' never changed to true, even though the Android Volvo On Call app reports that it is on.

The only difference I could find in the status output was this.

Engine on:
'ERS': {
'status': 'onByDirectCtrl',
'timestamp': datetime.datetime(2018, 9, 15, 15, 59, 6, tzinfo=datetime.timezone.utc), 
'engineStartWarning':'None',
'engineStartWarningTimestamp': datetime.datetime(2018, 9, 15, 15, 59, 6, tzinfo=datetime.timezone.utc)
}

Engine off (MaxAttemptsExceeded since I can only remote start 2 times before starting normally):
'ERS': {
'status':'off',
'timestamp': datetime.datetime(2018, 9, 15, 16, 17, 37, tzinfo=datetime.timezone.utc),
'engineStartWarning': 'MaxAttemptsExceeded',
'engineStartWarningTimestamp': datetime.datetime(2018, 9, 15, 16, 17, 37, tzinfo=datetime.timezone.utc)
}

So, ERS (engine remote start?) is probably being used instead of engineRunning.

**Comment by molobrakos** [comment]
Please test https://github.com/molobrakos/volvooncall/commit/18bfcfcb6e9ebfb0790dbe240da1fabfd718ba07

**Comment by TENFIRE** [comment]
Will test it now

**Comment by TENFIRE** [comment]
Seems like the dashboard does not display the correct value for the engine. Both dashboard and status printed "off" when the engine was off as it should.

Output (status/dashboard) with the engine running (remote started):
```
erik@Server-Erik:~/volvooncall$ ./voc status
<REMOVED> (XC60/2019) <REMOVED> 10224km (fuel 47% 390km)
    position: <REMOVED>, <REMOVED>
    engine: on
    locked: yes
    heater: off
erik@Server-Erik:~/volvooncall$ ./voc -i <REMOVED> dashboard
<REMOVED> Position: (<REMOVED>, <REMOVED>)
<REMOVED> Door lock: Locked
<REMOVED> Heater: Off
<REMOVED> Odometer: 10225 km
<REMOVED> Trip meter 1: 225 km
<REMOVED> Trip meter 2: 4 km
<REMOVED> Fuel amount: 34 L
<REMOVED> Fuel level: 47 %
<REMOVED> Fuel consumption: 8.0 L/100 km
<REMOVED> Range: 390 km
<REMOVED> Engine: Off
<REMOVED> Last trip: 2018-09-26 17:11:21+02:00
<REMOVED> Engine: Off
<REMOVED> Hood: Closed
<REMOVED> Front left door: Closed
<REMOVED> Front right door: Closed
<REMOVED> Rear left door: Closed
<REMOVED> Rear right door: Closed
<REMOVED> Front left window: Closed
<REMOVED> Front right window: Closed
<REMOVED> Rear left window: Closed
<REMOVED> Rear right window: Closed
18-09-26 18:11.57 ERROR (MainThread) [dashboard] Can not encode state None:<class 'NoneType'>
<REMOVED> Front right tyre: OK
18-09-26 18:11.57 ERROR (MainThread) [dashboard] Can not encode state None:<class 'NoneType'>
<REMOVED> Front left tyre: OK
18-09-26 18:11.57 ERROR (MainThread) [dashboard] Can not encode state None:<class 'NoneType'>
<REMOVED> Rear right tyre: OK
18-09-26 18:11.57 ERROR (MainThread) [dashboard] Can not encode state None:<class 'NoneType'>
<REMOVED> Rear left tyre: OK
<REMOVED> Washer fluid: OK
<REMOVED> Brake Fluid: OK
<REMOVED> Service: OK
<REMOVED> Bulbs: OK
<REMOVED> Doors: Closed
<REMOVED> Windows: Closed
```

**Comment by molobrakos** [comment]
heh, I guess we need a proper test suite with mocked server responses and expected results...

**Comment by molobrakos** [comment]
Like this: https://github.com/molobrakos/volvooncall/blob/master/test_main.py#L101

**Comment by molobrakos** [comment]
Let me know when you have tested this and I'll release a new version that can be used with Home Assistant.

**Comment by fluppie** [comment]
I just got my new 2019 V60 today, the Volvo on call plugin is working. Also want remote start in home assistant😊😊. This feature is supported by mine, so I can test this if needed.

**Comment by molobrakos** [comment]
Please do

**Comment by fluppie** [comment]
I replaced the volvooncall.py file from your Github but this give me some errors:
```
Log Details (ERROR)

Fri Oct 26 2018 23:40:30 GMT+0200 (Romance (zomertijd))

Error loading homeassistant.components.volvooncall. Make sure all dependencies are installed

Traceback (most recent call last):
  File "/srv/homeassistant/lib/python3.5/site-packages/homeassistant/loader.py", line 92, in get_component
    module = importlib.import_module(path)
  File "/usr/lib/python3.5/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 986, in _gcd_import
  File "<frozen importlib._bootstrap>", line 969, in _find_and_load
  File "<frozen importlib._bootstrap>", line 958, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 673, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 673, in exec_module
  File "<frozen importlib._bootstrap>", line 222, in _call_with_frames_removed
  File "/srv/homeassistant/lib/python3.5/site-packages/homeassistant/components/volvooncall.py", line 18, in <module>
    from util import obj_parser, json_serialize, is_valid_path, find_path
ImportError: No module named 'util'
```

**Comment by molobrakos** [comment]
Great, thanks! Primarily the stuff you reported here: https://github.com/molobrakos/volvooncall/issues/23#issuecomment-424779217
You can also test my Home Assistant branch here: https://github.com/molobrakos/home-assistant/tree/new-voc which contains the updated vocs-stuff.

**Comment by fluppie** [comment]
Can I just replace the files in my HA installation 0.80.3 where I see modifications?

**Comment by TENFIRE** [comment]
**Test results:**
`voc status:` Engine status is reported correctly

`voc dashboard:` Engine status is reported correctly

`homeassistant switch-on:` Engine is turned on, engine switch state is updated to "On". Minor delay (10-15 sec)
`homeassistant switch-off:` Engine is turned off, engine switch state is updated to "Off" after some delay (1-2 minutes)

**Comment by TENFIRE** [comment]
@fluppie I followed these instructions to get started:

https://developers.home-assistant.io/docs/en/development_environment.html

**Comment by fluppie** [comment]
OK, I made a dev environment and cloned from molobrakos github, it gives me a 0.77.0.dev0 Home Assistant env. Hereby the log:

> (home-assistant) pi@hassbian:~/home-assistant $ hass
Config directory: /home/pi/.homeassistant
2018-10-29 17:49:40 INFO (MainThread) [homeassistant.core] Bus:Handling <Event service_registered[L]: service=turn_off, domain=homeassistant>
2018-10-29 17:49:40 INFO (MainThread) [homeassistant.core] Bus:Handling <Event service_registered[L]: service=turn_on, domain=homeassistant>
2018-10-29 17:49:40 INFO (MainThread) [homeassistant.core] Bus:Handling <Event service_registered[L]: service=toggle, domain=homeassistant>
2018-10-29 17:49:40 INFO (MainThread) [homeassistant.core] Bus:Handling <Event service_registered[L]: service=stop, domain=homeassistant>
2018-10-29 17:49:40 INFO (MainThread) [homeassistant.core] Bus:Handling <Event service_registered[L]: service=restart, domain=homeassistant>
2018-10-29 17:49:40 INFO (MainThread) [homeassistant.core] Bus:Handling <Event service_registered[L]: service=check_config, domain=homeassistant>
2018-10-29 17:49:40 INFO (MainThread) [homeassistant.core] Bus:Handling <Event service_registered[L]: service=reload_core_config, domain=homeassistant>
2018-10-29 17:49:40 INFO (MainThread) [homeassistant.core] Bus:Handling <Event service_registered[L]: service=create, domain=persistent_notification>
2018-10-29 17:49:40 INFO (MainThread) [homeassistant.core] Bus:Handling <Event service_registered[L]: service=dismiss, domain=persistent_notification>
2018-10-29 17:49:40 INFO (MainThread) [homeassistant.bootstrap] Home Assistant core initialized
2018-10-29 17:49:40 INFO (MainThread) [homeassistant.loader] Loaded logger from homeassistant.components.logger
2018-10-29 17:49:40 INFO (MainThread) [homeassistant.setup] Setting up logger
2018-10-29 17:49:40 INFO (MainThread) [homeassistant.core] Bus:Handling <Event service_registered[L]: service=set_default_level, domain=logger>
2018-10-29 17:49:40 INFO (MainThread) [homeassistant.core] Bus:Handling <Event service_registered[L]: service=set_level, domain=logger>
2018-10-29 17:49:40 INFO (MainThread) [homeassistant.setup] Setup of domain logger took 0.0 seconds.
2018-10-29 17:49:40 INFO (MainThread) [homeassistant.core] Bus:Handling <Event component_loaded[L]: component=logger>
2018-10-29 17:49:41 INFO (MainThread) [homeassistant.loader] Loaded history from homeassistant.components.history
2018-10-29 17:49:41 INFO (MainThread) [homeassistant.loader] Loaded recorder from homeassistant.components.recorder
2018-10-29 17:49:41 INFO (MainThread) [homeassistant.loader] Loaded http from homeassistant.components.http
2018-10-29 17:49:41 INFO (MainThread) [homeassistant.loader] Loaded frontend from homeassistant.components.frontend
2018-10-29 17:49:41 INFO (MainThread) [homeassistant.loader] Loaded api from homeassistant.components.api
2018-10-29 17:49:41 INFO (MainThread) [homeassistant.loader] Loaded websocket_api from homeassistant.components.websocket_api
2018-10-29 17:49:41 INFO (MainThread) [homeassistant.loader] Loaded system_log from homeassistant.components.system_log
2018-10-29 17:49:41 INFO (MainThread) [homeassistant.loader] Loaded auth from homeassistant.components.auth
2018-10-29 17:49:41 INFO (MainThread) [homeassistant.loader] Loaded onboarding from homeassistant.components.onboarding
2018-10-29 17:49:41 INFO (MainThread) [homeassistant.loader] Loaded introduction from homeassistant.components.introduction
2018-10-29 17:49:41 INFO (MainThread) [homeassistant.setup] Setting up introduction
2018-10-29 17:49:41 INFO (MainThread) [homeassistant.components.introduction]

    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

        Hello, and welcome to Home Assistant!

        We'll hope that we can make all your dreams come true.

        Here are some resources to get started:

         - Configuring Home Assistant:
           https://home-assistant.io/getting-started/configuration/

         - Available components:
           https://home-assistant.io/components/

         - Troubleshooting your configuration:
           https://home-assistant.io/getting-started/troubleshooting-configuration/

         - Getting help:
           https://home-assistant.io/help/

        This message is generated by the introduction component. You can
        disable it in configuration.yaml.

    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

2018-10-29 17:49:41 INFO (MainThread) [homeassistant.loader] Loaded persistent_notification from homeassistant.components.persistent_notification
2018-10-29 17:49:41 INFO (MainThread) [homeassistant.setup] Setup of domain introduction took 0.0 seconds.
2018-10-29 17:49:41 INFO (MainThread) [homeassistant.core] Bus:Handling <Event component_loaded[L]: component=introduction>
2018-10-29 17:49:41 INFO (MainThread) [homeassistant.core] Bus:Handling <Event call_service[L]: service=create, service_data=message=
Here are some resources to get started:

 - [Configuring Home Assistant](https://home-assistant.io/getting-started/configuration/)
 - [Available components](https://home-assistant.io/components/)
 - [Troubleshooting your configuration](https://home-assistant.io/docs/configuration/troubleshooting/)
 - [Getting help](https://home-assistant.io/help/)

To not see this card popup in the future, edit your config in
`configuration.yaml` and disable the `introduction` component.
, title=Welcome Home!, domain=persistent_notification>
2018-10-29 17:49:41 INFO (MainThread) [homeassistant.core] Bus:Handling <Event state_changed[L]: old_state=None, entity_id=persistent_notification.notification, new_state=<state persistent_notification.notification=notifying; message=Here are some resources to get started:

 - [Configuring Home Assistant](https://home-assistant.io/getting-started/configuration/)
 - [Available components](https://home-assistant.io/components/)
 - [Troubleshooting your configuration](https://home-assistant.io/docs/configuration/troubleshooting/)
 - [Getting help](https://home-assistant.io/help/)

To not see this card popup in the future, edit your config in
`configuration.yaml` and disable the `introduction` component., title=Welcome Home! @ 2018-10-29T16:49:41.514335+00:00>>
2018-10-29 17:49:41 INFO (MainThread) [homeassistant.core] Bus:Handling <Event service_executed[L]>
2018-10-29 17:49:41 INFO (MainThread) [homeassistant.setup] Setting up http
2018-10-29 17:49:41 WARNING (MainThread) [homeassistant.components.http] You have been advised to set http.api_password.
2018-10-29 17:49:41 INFO (MainThread) [homeassistant.setup] Setup of domain http took 0.0 seconds.
2018-10-29 17:49:41 INFO (MainThread) [homeassistant.core] Bus:Handling <Event component_loaded[L]: component=http>
2018-10-29 17:49:41 INFO (MainThread) [homeassistant.setup] Setting up onboarding
2018-10-29 17:49:41 INFO (MainThread) [homeassistant.setup] Setting up api
2018-10-29 17:49:41 INFO (MainThread) [homeassistant.setup] Setting up websocket_api
2018-10-29 17:49:41 INFO (MainThread) [homeassistant.setup] Setup of domain websocket_api took 0.0 seconds.
2018-10-29 17:49:41 INFO (MainThread) [homeassistant.core] Bus:Handling <Event component_loaded[L]: component=websocket_api>
2018-10-29 17:49:41 INFO (MainThread) [homeassistant.setup] Setting up system_log
2018-10-29 17:49:41 INFO (MainThread) [homeassistant.core] Bus:Handling <Event service_registered[L]: service=clear, domain=system_log>
2018-10-29 17:49:41 INFO (MainThread) [homeassistant.core] Bus:Handling <Event service_registered[L]: service=write, domain=system_log>
2018-10-29 17:49:41 INFO (MainThread) [homeassistant.setup] Setup of domain system_log took 0.0 seconds.
2018-10-29 17:49:41 INFO (MainThread) [homeassistant.core] Bus:Handling <Event component_loaded[L]: component=system_log>
2018-10-29 17:49:41 INFO (MainThread) [homeassistant.setup] Setting up auth
2018-10-29 17:49:41 INFO (MainThread) [homeassistant.setup] Setup of domain auth took 0.0 seconds.
2018-10-29 17:49:41 INFO (MainThread) [homeassistant.core] Bus:Handling <Event component_loaded[L]: component=auth>
2018-10-29 17:49:41 INFO (MainThread) [homeassistant.setup] Setup of domain api took 0.0 seconds.
2018-10-29 17:49:41 INFO (MainThread) [homeassistant.core] Bus:Handling <Event component_loaded[L]: component=api>
2018-10-29 17:49:41 INFO (MainThread) [homeassistant.setup] Setup of domain onboarding took 0.1 seconds.
2018-10-29 17:49:41 INFO (MainThread) [homeassistant.core] Bus:Handling <Event component_loaded[L]: component=onboarding>
2018-10-29 17:49:41 INFO (MainThread) [homeassistant.setup] Setting up recorder
2018-10-29 17:49:41 INFO (MainThread) [homeassistant.core] Bus:Handling <Event service_registered[L]: service=purge, domain=recorder>
2018-10-29 17:49:42 INFO (MainThread) [homeassistant.setup] Setting up frontend
2018-10-29 17:49:43 INFO (MainThread) [homeassistant.setup] Setup of domain frontend took 1.6 seconds.
2018-10-29 17:49:43 INFO (MainThread) [homeassistant.core] Bus:Handling <Event component_loaded[L]: component=frontend>
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.setup] Setup of domain recorder took 2.3 seconds.
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.core] Bus:Handling <Event component_loaded[L]: component=recorder>
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.setup] Setting up history
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.setup] Setup of domain history took 0.0 seconds.
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.core] Bus:Handling <Event component_loaded[L]: component=history>
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.loader] Loaded sensor from homeassistant.components.sensor
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.loader] Loaded sensor.yr from homeassistant.components.sensor.yr
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.setup] Setting up sensor
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.loader] Loaded automation from homeassistant.components.automation
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.loader] Loaded group from homeassistant.components.group
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.loader] Loaded map from homeassistant.components.map
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.setup] Setting up map
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.setup] Setup of domain map took 0.0 seconds.
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.core] Bus:Handling <Event component_loaded[L]: component=map>
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.loader] Loaded config from homeassistant.components.config
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.setup] Setting up group
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.core] Bus:Handling <Event service_registered[L]: service=reload, domain=group>
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.core] Bus:Handling <Event service_registered[L]: service=set, domain=group>
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.core] Bus:Handling <Event service_registered[L]: service=remove, domain=group>
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.core] Bus:Handling <Event service_registered[L]: service=set_visibility, domain=group>
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.setup] Setup of domain group took 0.0 seconds.
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.core] Bus:Handling <Event component_loaded[L]: component=group>
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.loader] Loaded discovery from homeassistant.components.discovery
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.loader] Loaded logbook from homeassistant.components.logbook
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.loader] Loaded sun from homeassistant.components.sun
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.setup] Setting up sun
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.setup] Setup of domain sun took 0.3 seconds.
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.core] Bus:Handling <Event component_loaded[L]: component=sun>
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.loader] Loaded script from homeassistant.components.script
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.loader] Loaded conversation from homeassistant.components.conversation
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.loader] Loaded volvooncall from homeassistant.components.volvooncall
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.loader] Loaded tts from homeassistant.components.tts
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.loader] Loaded tts.google from homeassistant.components.tts.google
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.loader] Loaded cloud from homeassistant.components.cloud
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.loader] Loaded updater from homeassistant.components.updater
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.core] Bus:Handling <Event state_changed[L]: old_state=None, entity_id=sun.sun, new_state=<state sun.sun=above_horizon; next_dawn=2018-10-30T05:18:56+00:00, next_noon=2018-10-30T11:43:38+00:00, next_dusk=2018-10-29T18:08:22+00:00, next_rising=2018-10-30T05:40:13+00:00, next_midnight=2018-10-29T23:43:38+00:00, elevation=13.14, next_setting=2018-10-29T17:47:08+00:00, azimuth=255.8, friendly_name=Sun @ 2018-10-29T16:49:44.734889+00:00>>
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.setup] Setting up discovery
2018-10-29 17:49:44 ERROR (MainThread) [homeassistant.setup] Error during setup of component discovery
Traceback (most recent call last):
  File "/home/pi/home-assistant/homeassistant/setup.py", line 145, in _async_setup_component
    hass, processed_config)
  File "/home/pi/home-assistant/homeassistant/components/discovery.py", line 111, in async_setup
    from netdisco.discovery import NetworkDiscovery
  File "/home/pi/home-assistant/lib/python3.5/site-packages/netdisco/discovery.py", line 6, in <module>
    from .ssdp import SSDP
  File "/home/pi/home-assistant/lib/python3.5/site-packages/netdisco/ssdp.py", line 11, in <module>
    from netdisco.util import etree_to_dict, interface_addresses
  File "/home/pi/home-assistant/lib/python3.5/site-packages/netdisco/util.py", line 4, in <module>
    import netifaces
ImportError: No module named 'netifaces'
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.setup] Setting up automation
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.core] Bus:Handling <Event service_registered[L]: service=trigger, domain=automation>
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.core] Bus:Handling <Event service_registered[L]: service=reload, domain=automation>
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.core] Bus:Handling <Event service_registered[L]: service=toggle, domain=automation>
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.core] Bus:Handling <Event service_registered[L]: service=turn_on, domain=automation>
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.core] Bus:Handling <Event service_registered[L]: service=turn_off, domain=automation>
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.setup] Setup of domain automation took 0.0 seconds.
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.core] Bus:Handling <Event component_loaded[L]: component=automation>
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.setup] Setting up config
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.setup] Setting up logbook
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.core] Bus:Handling <Event service_registered[L]: service=log, domain=logbook>
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.setup] Setup of domain logbook took 0.0 seconds.
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.core] Bus:Handling <Event component_loaded[L]: component=logbook>
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.core] Bus:Handling <Event call_service[L]: service=create, service_data=message=The following components and platforms could not be set up:

 - [discovery](https://home-assistant.io/components/discovery/)

Please check your config., title=Invalid config, notification_id=invalid_config, domain=persistent_notification>
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.setup] Setting up script
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.core] Bus:Handling <Event service_registered[L]: service=reload, domain=script>
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.core] Bus:Handling <Event service_registered[L]: service=turn_on, domain=script>
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.core] Bus:Handling <Event service_registered[L]: service=turn_off, domain=script>
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.core] Bus:Handling <Event service_registered[L]: service=toggle, domain=script>
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.setup] Setup of domain script took 0.0 seconds.
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.core] Bus:Handling <Event component_loaded[L]: component=script>
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.setup] Setting up conversation
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.core] Bus:Handling <Event service_registered[L]: service=process, domain=conversation>
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.setup] Setup of domain conversation took 0.0 seconds.
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.core] Bus:Handling <Event component_loaded[L]: component=conversation>
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.loader] Loaded config.automation from homeassistant.components.config.automation
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.core] Bus:Handling <Event component_loaded[L]: component=config.automation>
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.loader] Loaded config.config_entries from homeassistant.components.config.config_entries
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.loader] Loaded config.hassbian from homeassistant.components.config.hassbian
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.loader] Loaded config.entity_registry from homeassistant.components.config.entity_registry
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.loader] Loaded config.core from homeassistant.components.config.core
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.core] Bus:Handling <Event component_loaded[L]: component=config.core>
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.loader] Loaded config.customize from homeassistant.components.config.customize
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.core] Bus:Handling <Event component_loaded[L]: component=config.customize>
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.loader] Loaded config.script from homeassistant.components.config.script
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.core] Bus:Handling <Event component_loaded[L]: component=config.script>
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.loader] Loaded config.group from homeassistant.components.config.group
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.core] Bus:Handling <Event component_loaded[L]: component=config.group>
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.core] Bus:Handling <Event state_changed[L]: old_state=None, entity_id=persistent_notification.invalid_config, new_state=<state persistent_notification.invalid_config=notifying; message=The following components and platforms could not be set up:

 - [discovery](https://home-assistant.io/components/discovery/)

Please check your config., title=Invalid config @ 2018-10-29T16:49:44.927425+00:00>>
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.core] Bus:Handling <Event service_executed[L]>
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.core] Bus:Handling <Event component_loaded[L]: component=config.entity_registry>
2018-10-29 17:49:44 INFO (MainThread) [homeassistant.setup] Setting up volvooncall
2018-10-29 17:49:44 INFO (Thread-8) [volvooncall] Initializing volvooncall version: 0.6.11
2018-10-29 17:49:44 INFO (Thread-8) [homeassistant.components.volvooncall] Logging in to service
2018-10-29 17:49:44 INFO (Thread-8) [volvooncall] Updating
2018-10-29 17:49:44 INFO (Thread-8) [volvooncall] Querying vehicles
2018-10-29 17:49:45 INFO (MainThread) [homeassistant.setup] Setting up updater
2018-10-29 17:49:45 WARNING (MainThread) [homeassistant.components.updater] Running on 'dev', only analytics will be submitted
2018-10-29 17:49:45 INFO (MainThread) [homeassistant.setup] Setup of domain updater took 0.0 seconds.
2018-10-29 17:49:45 INFO (MainThread) [homeassistant.core] Bus:Handling <Event component_loaded[L]: component=updater>
2018-10-29 17:49:45 INFO (MainThread) [homeassistant.components.sensor] Setting up sensor.yr
2018-10-29 17:49:45 INFO (MainThread) [homeassistant.core] Bus:Handling <Event state_changed[L]: old_state=None, entity_id=sensor.yr_symbol, new_state=<state sensor.yr_symbol=unknown; friendly_name=yr Symbol, entity_picture=https://api.met.no/weatherapi/weathericon/1.1/?symbol=None;content_type=image/png, attribution=Weather forecast from met.no, delivered by the Norwegian Meteorological Institute. @ 2018-10-29T16:49:45.290888+00:00>>
2018-10-29 17:49:45 INFO (MainThread) [homeassistant.setup] Setting up tts
2018-10-29 17:49:45 INFO (MainThread) [homeassistant.setup] Setting up cloud
2018-10-29 17:49:45 INFO (MainThread) [homeassistant.setup] Setup of domain cloud took 0.0 seconds.
2018-10-29 17:49:45 INFO (MainThread) [homeassistant.core] Bus:Handling <Event component_loaded[L]: component=cloud>
2018-10-29 17:49:45 INFO (MainThread) [homeassistant.core] Bus:Handling <Event component_loaded[L]: component=config.config_entries>
2018-10-29 17:49:45 INFO (MainThread) [homeassistant.setup] Setup of domain config took 0.7 seconds.
2018-10-29 17:49:45 INFO (MainThread) [homeassistant.core] Bus:Handling <Event component_loaded[L]: component=config>
2018-10-29 17:49:45 INFO (MainThread) [homeassistant.core] Bus:Handling <Event state_changed[L]: old_state=<state sensor.yr_symbol=unknown; friendly_name=yr Symbol, entity_picture=https://api.met.no/weatherapi/weathericon/1.1/?symbol=None;content_type=image/png, attribution=Weather forecast from met.no, delivered by the Norwegian Meteorological Institute. @ 2018-10-29T16:49:45.290888+00:00>, entity_id=sensor.yr_symbol, new_state=<state sensor.yr_symbol=2; friendly_name=yr Symbol, entity_picture=https://api.met.no/weatherapi/weathericon/1.1/?symbol=2;content_type=image/png, attribution=Weather forecast from met.no, delivered by the Norwegian Meteorological Institute. @ 2018-10-29T16:49:45.870332+00:00>>
2018-10-29 17:49:45 INFO (MainThread) [homeassistant.setup] Setup of domain sensor took 1.7 seconds.
2018-10-29 17:49:45 INFO (MainThread) [homeassistant.core] Bus:Handling <Event component_loaded[L]: component=sensor>
2018-10-29 17:49:45 INFO (MainThread) [homeassistant.core] Bus:Handling <Event service_registered[L]: service=google_say, domain=tts>
2018-10-29 17:49:45 INFO (MainThread) [homeassistant.core] Bus:Handling <Event service_registered[L]: service=clear_cache, domain=tts>
2018-10-29 17:49:45 INFO (MainThread) [homeassistant.setup] Setup of domain tts took 0.6 seconds.
2018-10-29 17:49:45 INFO (MainThread) [homeassistant.core] Bus:Handling <Event component_loaded[L]: component=tts>
2018-10-29 17:49:46 ERROR (MainThread) [homeassistant.setup] Error during setup of component volvooncall
Traceback (most recent call last):
  File "/home/pi/home-assistant/homeassistant/setup.py", line 148, in _async_setup_component
    component.setup, hass, processed_config)  # type: ignore
  File "/usr/lib/python3.5/asyncio/futures.py", line 380, in __iter__
    yield self  # This tells Task to wait for completion.
  File "/usr/lib/python3.5/asyncio/tasks.py", line 304, in _wakeup
    future.result()
  File "/usr/lib/python3.5/asyncio/futures.py", line 293, in result
    raise self._exception
  File "/usr/lib/python3.5/concurrent/futures/thread.py", line 55, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/home/pi/home-assistant/homeassistant/components/volvooncall.py", line 104, in setup
    return update(utcnow())
  File "/home/pi/home-assistant/homeassistant/components/volvooncall.py", line 91, in update
    discover_vehicle(vehicle)
  File "/home/pi/home-assistant/homeassistant/components/volvooncall.py", line 73, in discover_vehicle
    dashboard = vehicle.dashboard
  File "/home/pi/home-assistant/lib/python3.5/site-packages/volvooncall.py", line 394, in dashboard
    from dashboard import Dashboard
  File "/home/pi/home-assistant/lib/python3.5/site-packages/dashboard.py", line 45
    return f'{self.vehicle_name} {self.name}'
                                            ^
SyntaxError: invalid syntax
2018-10-29 17:49:46 INFO (MainThread) [homeassistant.core] Bus:Handling <Event call_service[L]: service=create, service_data=message=The following components and platforms could not be set up:

 - [discovery](https://home-assistant.io/components/discovery/)
 - [volvooncall](https://home-assistant.io/components/volvooncall/)

Please check your config., title=Invalid config, notification_id=invalid_config, domain=persistent_notification>
2018-10-29 17:49:46 INFO (MainThread) [homeassistant.core] Bus:Handling <Event state_changed[L]: old_state=<state persistent_notification.invalid_config=notifying; message=The following components and platforms could not be set up:

 - [discovery](https://home-assistant.io/components/discovery/)

Please check your config., title=Invalid config @ 2018-10-29T16:49:44.927425+00:00>, entity_id=persistent_notification.invalid_config, new_state=<state persistent_notification.invalid_config=notifying; message=The following components and platforms could not be set up:

 - [discovery](https://home-assistant.io/components/discovery/)
 - [volvooncall](https://home-assistant.io/components/volvooncall/)

Please check your config., title=Invalid config @ 2018-10-29T16:49:44.927425+00:00>>
2018-10-29 17:49:46 INFO (MainThread) [homeassistant.core] Bus:Handling <Event service_executed[L]>
2018-10-29 17:49:46 INFO (MainThread) [homeassistant.bootstrap] Home Assistant initialized in 5.64s
2018-10-29 17:49:46 INFO (MainThread) [homeassistant.core] Starting Home Assistant core loop
2018-10-29 17:49:46 INFO (MainThread) [homeassistant.core] Starting Home Assistant
2018-10-29 17:49:46 INFO (MainThread) [homeassistant.core] Bus:Handling <Event homeassistant_start[L]>
2018-10-29 17:49:46 INFO (MainThread) [homeassistant.core] Timer:starting
2018-10-29 17:50:30 INFO (MainThread) [homeassistant.core] Bus:Handling <Event state_changed[L]: old_state=<state sun.sun=above_horizon; next_dawn=2018-10-30T05:18:56+00:00, next_noon=2018-10-30T11:43:38+00:00, next_dusk=2018-10-29T18:08:22+00:00, next_rising=2018-10-30T05:40:13+00:00, next_midnight=2018-10-29T23:43:38+00:00, elevation=13.14, next_setting=2018-10-29T17:47:08+00:00, azimuth=255.8, friendly_name=Sun @ 2018-10-29T16:49:44.734889+00:00>, entity_id=sun.sun, new_state=<state sun.sun=above_horizon; next_dawn=2018-10-30T05:18:56+00:00, next_noon=2018-10-30T11:43:38+00:00, next_dusk=2018-10-29T18:08:22+00:00, next_rising=2018-10-30T05:40:13+00:00, next_midnight=2018-10-29T23:43:38+00:00, elevation=12.96, next_setting=2018-10-29T17:47:08+00:00, azimuth=255.82, friendly_name=Sun @ 2018-10-29T16:49:44.734889+00:00>>
2018-10-29 17:50:37 INFO (MainThread) [homeassistant.components.http.view] Serving /api/websocket to 94.226.162.4 (auth: True)
2018-10-29 17:50:37 INFO (MainThread) [homeassistant.loader] Loaded homematicip_cloud from homeassistant.components.homematicip_cloud
2018-10-29 17:50:37 INFO (MainThread) [homeassistant.loader] Loaded deconz from homeassistant.components.deconz
2018-10-29 17:50:37 INFO (MainThread) [homeassistant.loader] Loaded sonos from homeassistant.components.sonos
2018-10-29 17:50:37 INFO (MainThread) [homeassistant.loader] Loaded nest from homeassistant.components.nest
2018-10-29 17:50:37 INFO (MainThread) [homeassistant.loader] Loaded zone from homeassistant.components.zone
2018-10-29 17:50:37 INFO (MainThread) [homeassistant.loader] Loaded hue from homeassistant.components.hue
2018-10-29 17:50:37 INFO (MainThread) [homeassistant.loader] Loaded cast from homeassistant.components.cast
2018-10-29 17:50:37 INFO (MainThread) [homeassistant.components.http.view] Serving /api/error/all to 94.226.162.4 (auth: True)
2018-10-29 17:50:46 INFO (Thread-18) [volvooncall] Updating
2018-10-29 17:51:05 INFO (MainThread) [homeassistant.components.http.view] Serving /api/states/persistent_notification.notification to 94.226.162.4 (auth: True)
2018-10-29 17:51:05 INFO (MainThread) [homeassistant.core] Bus:Handling <Event state_changed[L]: old_state=<state persistent_notification.notification=notifying; message=Here are some resources to get started:

 - [Configuring Home Assistant](https://home-assistant.io/getting-started/configuration/)
 - [Available components](https://home-assistant.io/components/)
 - [Troubleshooting your configuration](https://home-assistant.io/docs/configuration/troubleshooting/)
 - [Getting help](https://home-assistant.io/help/)

To not see this card popup in the future, edit your config in
`configuration.yaml` and disable the `introduction` component., title=Welcome Home! @ 2018-10-29T16:49:41.514335+00:00>, entity_id=persistent_notification.notification, new_state=None>
2018-10-29 17:51:12 INFO (MainThread) [homeassistant.components.http.view] Serving /api/states/persistent_notification.invalid_config to 94.226.162.4 (auth: True)
2018-10-29 17:51:12 INFO (MainThread) [homeassistant.core] Bus:Handling <Event state_changed[L]: old_state=<state persistent_notification.invalid_config=notifying; message=The following components and platforms could not be set up:

 - [discovery](https://home-assistant.io/components/discovery/)
 - [volvooncall](https://home-assistant.io/components/volvooncall/)

Please check your config., title=Invalid config @ 2018-10-29T16:49:44.927425+00:00>, entity_id=persistent_notification.invalid_config, new_state=None>
2018-10-29 17:51:16 INFO (MainThread) [homeassistant.components.http.view] Serving /api/cloud/account to 94.226.162.4 (auth: True)
2018-10-29 17:51:17 INFO (MainThread) [homeassistant.components.http.view] Serving /api/error/all to 94.226.162.4 (auth: True)
2018-10-29 17:51:30 INFO (MainThread) [homeassistant.core] Bus:Handling <Event state_changed[L]: old_state=<state sun.sun=above_horizon; next_dawn=2018-10-30T05:18:56+00:00, next_noon=2018-10-30T11:43:38+00:00, next_dusk=2018-10-29T18:08:22+00:00, next_rising=2018-10-30T05:40:13+00:00, next_midnight=2018-10-29T23:43:38+00:00, elevation=12.96, next_setting=2018-10-29T17:47:08+00:00, azimuth=255.82, friendly_name=Sun @ 2018-10-29T16:49:44.734889+00:00>, entity_id=sun.sun, new_state=<state sun.sun=above_horizon; next_dawn=2018-10-30T05:18:56+00:00, next_noon=2018-10-30T11:43:38+00:00, next_dusk=2018-10-29T18:08:22+00:00, next_rising=2018-10-30T05:40:13+00:00, next_midnight=2018-10-29T23:43:38+00:00, elevation=12.72, next_setting=2018-10-29T17:47:08+00:00, azimuth=255.83, friendly_name=Sun @ 2018-10-29T16:49:44.734889+00:00>>
2018-10-29 17:51:39 INFO (MainThread) [homeassistant.components.http.view] Serving /api/error/all to 94.226.162.4 (auth: True)
2018-10-29 17:51:44 INFO (MainThread) [homeassistant.components.http.view] Serving /api/error_log to 94.226.162.4 (auth: True)
2018-10-29 17:51:47 INFO (Thread-19) [volvooncall] Updating
2018-10-29 17:52:30 INFO (MainThread) [homeassistant.core] Bus:Handling <Event state_changed[L]: old_state=<state sun.sun=above_horizon; next_dawn=2018-10-30T05:18:56+00:00, next_noon=2018-10-30T11:43:38+00:00, next_dusk=2018-10-29T18:08:22+00:00, next_rising=2018-10-30T05:40:13+00:00, next_midnight=2018-10-29T23:43:38+00:00, elevation=12.72, next_setting=2018-10-29T17:47:08+00:00, azimuth=255.83, friendly_name=Sun @ 2018-10-29T16:49:44.734889+00:00>, entity_id=sun.sun, new_state=<state sun.sun=above_horizon; next_dawn=2018-10-30T05:18:56+00:00, next_noon=2018-10-30T11:43:38+00:00, next_dusk=2018-10-29T18:08:22+00:00, next_rising=2018-10-30T05:40:13+00:00, next_midnight=2018-10-29T23:43:38+00:00, elevation=12.48, next_setting=2018-10-29T17:47:08+00:00, azimuth=255.84, friendly_name=Sun @ 2018-10-29T16:49:44.734889+00:00>>
2018-10-29 17:52:48 INFO (Thread-11) [volvooncall] Updating
2018-10-29 17:53:30 INFO (MainThread) [homeassistant.core] Bus:Handling <Event state_changed[L]: old_state=<state sun.sun=above_horizon; next_dawn=2018-10-30T05:18:56+00:00, next_noon=2018-10-30T11:43:38+00:00, next_dusk=2018-10-29T18:08:22+00:00, next_rising=2018-10-30T05:40:13+00:00, next_midnight=2018-10-29T23:43:38+00:00, elevation=12.48, next_setting=2018-10-29T17:47:08+00:00, azimuth=255.84, friendly_name=Sun @ 2018-10-29T16:49:44.734889+00:00>, entity_id=sun.sun, new_state=<state sun.sun=above_horizon; next_dawn=2018-10-30T05:18:56+00:00, next_noon=2018-10-30T11:43:38+00:00, next_dusk=2018-10-29T18:08:22+00:00, next_rising=2018-10-30T05:40:13+00:00, next_midnight=2018-10-29T23:43:38+00:00, elevation=12.24, next_setting=2018-10-29T17:47:08+00:00, azimuth=255.85, friendly_name=Sun @ 2018-10-29T16:49:44.734889+00:00>>

**Comment by fluppie** [comment]
I tried again on a RPi system without HA:

> 
2018-10-30 13:26:08 INFO (MainThread) [homeassistant.setup] Setup of domain sensor took 1.7 seconds.
2018-10-30 13:26:08 INFO (MainThread) [homeassistant.core] Bus:Handling <Event component_loaded[L]: component=sensor>
2018-10-30 13:26:08 ERROR (MainThread) [homeassistant.setup] Error during setup of component volvooncall
Traceback (most recent call last):
  File "/home/pi/home-assistant/homeassistant/setup.py", line 148, in _async_setup_component
    component.setup, hass, processed_config)  # type: ignore
  File "/usr/lib/python3.5/asyncio/futures.py", line 380, in __iter__
    yield self  # This tells Task to wait for completion.
  File "/usr/lib/python3.5/asyncio/tasks.py", line 304, in _wakeup
    future.result()
  File "/usr/lib/python3.5/asyncio/futures.py", line 293, in result
    raise self._exception
  File "/usr/lib/python3.5/concurrent/futures/thread.py", line 55, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/home/pi/home-assistant/homeassistant/components/volvooncall.py", line 104, in setup
    return update(utcnow())
  File "/home/pi/home-assistant/homeassistant/components/volvooncall.py", line 91, in update
    discover_vehicle(vehicle)
  File "/home/pi/home-assistant/homeassistant/components/volvooncall.py", line 73, in discover_vehicle
    dashboard = vehicle.dashboard
  File "/home/pi/home-assistant/lib/python3.5/site-packages/volvooncall.py", line 394, in dashboard
    from dashboard import Dashboard
  File "/home/pi/home-assistant/lib/python3.5/site-packages/dashboard.py", line 45
    return f'{self.vehicle_name} {self.name}'
                                            ^
SyntaxError: invalid syntax
2018-10-30 13:26:08 INFO (MainThread) [homeassistant.core] Bus:Handling <Event call_service[L]: domain=persistent_notification, service_data=title=Invalid config, notification_id=invalid_config, message=The following components and platforms could not be set up:

 - [volvooncall](https://home-assistant.io/components/volvooncall/)
 - [discovery](https://home-assistant.io/components/discovery/)

When using git checkout master everyting works. So only the failure when using git checkout new-voc

**Comment by molobrakos** [comment]
@fluppie, yes your Python version is too old.

**Comment by fluppie** [comment]
Working :) :)!!

![afbeelding](https://user-images.githubusercontent.com/8402106/47788076-fbe36880-dd10-11e8-9a29-f7d1c18ecbe0.png)

**Comment by molobrakos** [comment]
> **Test results:**
> `voc status:` Engine status is reported correctly
> 
> `voc dashboard:` Engine status is reported correctly

Great!
 
> `homeassistant switch-on:` Engine is turned on, engine switch state is updated to "On". Minor delay (10-15 sec)
> `homeassistant switch-off:` Engine is turned off, engine switch state is updated to "Off" after some delay (1-2 minutes)

Ok, I am not sure about the delay, but I think I will leave this for now. Maybe we can either set engine to on/off optimistically after executing each command. Or maybe we need to trigger an extra update from the server. Or maybe we have to wait for the server to update it's state form the vehicle first.

**Comment by fluppie** [comment]
Start/stop the engine is within 10s, most of the time 8.5s
Engine status update 1 - 1.5 minutes
Mine shows the heater, but I don't have that option on mine, it's also listed in the VOC app but gives an error after using it.

Can we also specify the timer for remote start? I think through the Volvo app we can select 1 to 15 min.

```
(home-assistant) pi@raspimqtt:~/home-assistant $ voc status -u xxxxx -p xxxxx
None (V60/2019) YV1ZWA00000000000 538km (fuel 20% 110km)
    position: 51.16406250000000,4.42350006103516
    engine: off
    locked: yes
    heater: off
(home-assistant) pi@raspimqtt:~/home-assistant $ voc dashboard -u xxxxx -p xxxxx
YV1ZWA00000000000 Position: (51.1640625, 4.423500061035156)
YV1ZWA00000000000 Door lock: Locked
YV1ZWA00000000000 Heater: Off
YV1ZWA00000000000 Odometer: 538 km
YV1ZWA00000000000 Trip meter 1: 538 km
YV1ZWA00000000000 Trip meter 2: 26 km
YV1ZWA00000000000 Fuel amount: 11 L
YV1ZWA00000000000 Fuel level: 20 %
YV1ZWA00000000000 Fuel consumption: 9.8 L/100 km
YV1ZWA00000000000 Range: 110 km
YV1ZWA00000000000 Engine: Off
YV1ZWA00000000000 Engine: Off
YV1ZWA00000000000 Hood: Closed
YV1ZWA00000000000 Front left door: Closed
YV1ZWA00000000000 Front right door: Closed
YV1ZWA00000000000 Rear left door: Closed
YV1ZWA00000000000 Rear right door: Closed
YV1ZWA00000000000 Front left window: Closed
YV1ZWA00000000000 Front right window: Closed
YV1ZWA00000000000 Rear left window: Closed
YV1ZWA00000000000 Rear right window: Closed
YV1ZWA00000000000 Front right tyre: OK
YV1ZWA00000000000 Front left tyre: OK
YV1ZWA00000000000 Rear right tyre: OK
YV1ZWA00000000000 Rear left tyre: OK
YV1ZWA00000000000 Washer fluid: OK
YV1ZWA00000000000 Brake Fluid: OK
YV1ZWA00000000000 Service: OK
YV1ZWA00000000000 Bulbs: OK
YV1ZWA00000000000 Doors: Closed
YV1ZWA00000000000 Windows: Closed
```

**Comment by molobrakos** [comment]
@fluppie if you do `voc print`and search for `preclimatizationSupported`, `remoteClimatizationStatus`, `remoteHeaterSupported`, and `heater` - what does it say for each item?

**Comment by fluppie** [comment]
"preclimatizationSupported": true,
    "registrationNumber": null,
    "remoteClimatizationStatus": null,
    "remoteClimatizationStatusTimestamp": null,
    "remoteHeaterSupported": false,
    "sendPOIToVehicleVersionsSupported": [
        "sendSinglePOIToVehicle"
    ],
    "serverVersion": "4.19.73-feature-CR_Fleet-voc-deployable",
    "serviceWarningStatus": "Normal",
    "serviceWarningStatusTimestamp": "2018-10-31T12:59:53+00:00",
    "status.parkedIndoor.supported": false,
    "subscriptionEndDate": "2021-10-26T00:00:00+00:00",
    "subscriptionStartDate": "2018-10-26T00:00:00+00:00",
    "subscriptionType": "PREMIUM_2013",
    "theftAlarm": null,
    "timeFullyAccessible": 7200,
    "timeFullyAccessibleUntil": "2018-11-05T12:51:45+00:00",
    "timePartiallyAccessible": 31680,
    "timePartiallyAccessibleUntil": "2018-11-22T12:51:45+00:00",
    "tripMeter1": 538300,
    "tripMeter1Timestamp": "2018-10-31T12:59:53+00:00",
    "tripMeter2": 26100,
    "tripMeter2Timestamp": "2018-10-31T12:59:53+00:00",
    "tyreDimensionCode": "2441R517",
    "tyreInflationPressureHeavyCode": null,
    "tyreInflationPressureLightCode": null,
    "tyrePressure": {
        "frontLeftTyrePressure": null,
        "frontRightTyrePressure": null,
        "rearLeftTyrePressure": null,
        "rearRightTyrePressure": null,
        "timestamp": null

**Comment by fluppie** [comment]
```
    def start_engine(self):
        if self.is_engine_start_supported:
            self.call('engine/start', runtime=5)
self.update()
```
Can we make the runtime dynamic through a slider in Home Assistant? Or just default to 15min? I would think most people like to run it for 15min during wintertime?

**Comment by molobrakos** [comment]
> Can we add support for Blink and Honk&Blink?

Sure we can, feel free to submit a PR for this. Previous discussion in https://github.com/molobrakos/volvooncall/issues/19

**Comment by molobrakos** [comment]
> Can we make the runtime dynamic through a slider in Home Assistant? Or just default to 15min? I would think most people like to run it for 15min during wintertime?
> 
> I think the brightness slider from lights would be a good approach?

Feel free to submit a PR for this (maybe a custom Home Assistant user interface-item has to be made as well).

**Comment by molobrakos** [comment]
> Mine shows the heater, but I don't have that option on mine, it's also listed in the VOC app but gives an error after using it.

My car only have support for the heater so I can not debug this. For previous discussion regarding heater vs preclimatization, please see: https://github.com/molobrakos/volvooncall/pull/2 I am not sure what to do with this. Maybe we should separate preclimatization from heater entirely and provide two different switches, but in your case it would not work anyway, since your car does report `preclimatizationSupported: true`.

**Comment by fluppie** [comment]
Ok weird, was making a new setup to modify code and try new stuff. The steps I've done to make a working environment don't work anymore. Did meanwhile something change?

> 2018-11-01 11:16:56 INFO (MainThread) [homeassistant.core] Bus:Handling <Event service_executed[L]>
> 2018-11-01 11:16:56 INFO (SyncWorker_6) [homeassistant.util.package] Attempting install of volvooncall==0.6.12
> 2018-11-01 11:17:00 ERROR (SyncWorker_6) [homeassistant.util.package] Unable to install package volvooncall==0.6.12: Could not find a version that satisfies the requirement volvooncall==0.6.12 (from versions: 0.1.1, 0.1.2, 0.1.5, 0.1.6, 0.1.7, 0.1.8, 0.1.9, 0.1.10, 0.2.0, 0.3.0, 0.3.1, 0.3.2, 0.3.3, 0.4.0, 0.4.1, 0.4.2, 0.4.4, 0.5.0, 0.6.0, 0.6.1, 0.6.2, 0.6.3, 0.6.4, 0.6.5, 0.6.6, 0.6.7, 0.6.8, 0.6.9, 0.6.11)
> No matching distribution found for volvooncall==0.6.12
> 2018-11-01 11:17:00 ERROR (MainThread) [homeassistant.requirements] Not initializing volvooncall because could not install requirement volvooncall==0.6.12
> 2018-11-01 11:17:00 ERROR (MainThread) [homeassistant.setup] Setup failed for volvooncall: Could not install all requirements.
>

**Comment by molobrakos** [comment]
@fluppie sorry, something seems to have failed with 0.6.12. Please pull and test again.

**Comment by fluppie** [comment]
> remote: Enumerating objects: 6, done.
> remote: Counting objects: 100% (6/6), done.
> remote: Compressing objects: 100% (2/2), done.
> remote: Total 6 (delta 4), reused 6 (delta 4), pack-reused 0
> Unpacking objects: 100% (6/6), done.
> From https://github.com/molobrakos/home-assistant
>  + dd84eceea...7bb79abc3 new-voc    -> origin/new-voc  (forced update)
> Auto-merging requirements_all.txt
> CONFLICT (content): Merge conflict in requirements_all.txt
> Auto-merging homeassistant/components/volvooncall.py
> CONFLICT (content): Merge conflict in homeassistant/components/volvooncall.py
> Automatic merge failed; fix conflicts and then commit the result.

Launching `hass`

> 2018-11-01 11:41:42 ERROR (MainThread) [homeassistant.core] Error doing job: Task exception was never retrieved
> Traceback (most recent call last):
>   File "/home/pi/home-assistant/homeassistant/setup.py", line 56, in async_setup_component
>     return await task  # type: ignore
>   File "/home/pi/home-assistant/homeassistant/setup.py", line 103, in _async_setup_component
>     component = loader.get_component(hass, domain)
>   File "/home/pi/home-assistant/homeassistant/loader.py", line 94, in get_component
>     module = importlib.import_module(path)
>   File "/usr/local/lib/python3.6/importlib/__init__.py", line 126, in import_module
>     return _bootstrap._gcd_import(name[level:], package, level)
>   File "<frozen importlib._bootstrap>", line 994, in _gcd_import
>   File "<frozen importlib._bootstrap>", line 971, in _find_and_load
>   File "<frozen importlib._bootstrap>", line 955, in _find_and_load_unlocked
>   File "<frozen importlib._bootstrap>", line 665, in _load_unlocked
>   File "<frozen importlib._bootstrap_external>", line 674, in exec_module
>   File "<frozen importlib._bootstrap_external>", line 781, in get_code
>   File "<frozen importlib._bootstrap_external>", line 741, in source_to_code
>   File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
>   File "/home/pi/home-assistant/homeassistant/components/volvooncall.py", line 25
>     <<<<<<< HEAD
>      ^
> SyntaxError: invalid syntax

Then I did `git reset --hard origin/new-voc
`
And all was fine

**Comment by molobrakos** [comment]
Hopefully works in Home Assistant dev-branch now, testing welcome!

**Comment by TENFIRE** [comment]
Hi, I have tested the homeassistant dev branch now.

**I'll start with the state of the switch.**
While parked (remote start off): switch-state = off
While parked (remote start on): switch-state = on
While driving: switch-state = on

So far so good, **however when I try to start/stop remotestart via the switch nothing happens**, I did not see anything in my logs either. :(

**Comment by TENFIRE** [comment]
> Can we also specify the timer for remote start? I think through the Volvo app we can select 1 to 15 min.
> 

I agree with @fluppie that 15 minutes is probably more desirable as a default value since 5 minutes is not enough to defrost windows. I have created a pull request for this #37

**Comment by molobrakos** [comment]
> So far so good, **however when I try to start/stop remotestart via the switch nothing happens**, I did not see anything in my logs either. :(

Does the standalone `voc -vv engine start` work?

**Comment by TENFIRE** [comment]
> > So far so good, **however when I try to start/stop remotestart via the switch nothing happens**, I did not see anything in my logs either. :(
> 
> Does the standalone `voc -vv engine start` work?

No, I got this:
`./voc:229: RuntimeWarning: coroutine 'Vehicle.start_engine' was never awaited`

**Comment by TENFIRE** [comment]
Tested the heater as well now, got the same result as for the engine

**Comment by molobrakos** [comment]
Ah, this commit fixes it: https://github.com/molobrakos/volvooncall/commit/9aeda75fc2c0895c6a7860677d075f46bc156f3e

**Comment by TENFIRE** [comment]
Got this now:
`18-12-03 17:12.34 WARNING (MainThread) [volvooncall] Failure to execute: _request() missing 1 required positional argument: 'ref'`

**Comment by molobrakos** [comment]
Thank you so much for your patience! How about this: https://github.com/molobrakos/volvooncall/commit/2663881e89b7826a6757df5090184a213eb472d5

**Comment by TENFIRE** [comment]
Ok, I got a little bit further this time:

```
18-12-03 20:57.41 DEBUG (MainThread) [volvooncall] Request for https://vocapi.wirelesscar.net/customerapi/rest/v3.0/engine/start
18-12-03 20:57.41 WARNING (MainThread) [volvooncall] Failure when communcating with the server: _request() got an unexpected keyword argument 'runtime'
18-12-03 20:57.41 WARNING (MainThread) [volvooncall] Failure to execute: _request() got an unexpected keyword argument 'runtime'
```

**Comment by molobrakos** [comment]
@fluppie, I wish I could blame the online service, but in this case it was my fault :-/ 
Soon there now, @TENFIRE :-) https://github.com/molobrakos/volvooncall/commit/8f117b1610fa4bef97d2995611836dd1dc0c39b3

**Comment by TENFIRE** [comment]
> @fluppie, I wish I could blame the online service, but in this case it was my fault :-/
> Soon there now, @TENFIRE :-) [8f117b1](https://github.com/molobrakos/volvooncall/commit/8f117b1610fa4bef97d2995611836dd1dc0c39b3)

Both `./voc -vv engine start` and `./voc -vv engine stop` works now. 👍 

Only DEBUG and INFO rows logged.

**Comment by fluppie** [comment]
I did a git pull from your HA fork, activated the seperate python virtual env and did script/setup and then hass. On first sight it works, I see the data of the Volvo etc. Only the command engine start does not work.

> 2018-12-05 17:57:51 WARNING (MainThread) [volvooncall] Failure to execute: _request() missing 1 required positional argument: 'ref'
2018-12-05 17:58:00 DEBUG (MainThread) [homeassistant.core] Bus:Handling <Event call_service[L]: service_data=entity_id=['switch.yv1XXXXXXXXXXXXXX_engine'], domain=switch, service=turn_on>
2018-12-05 17:58:00 WARNING (MainThread) [volvooncall] Failure to execute: _request() missing 1 required positional argument: 'ref'
2018-12-05 17:58:03 INFO (MainThread) [volvooncall] Updating

>

**Comment by fluppie** [comment]
> git checkout voc

and all is fine
![voc](https://user-images.githubusercontent.com/8402106/49530730-1cc34e80-f8b9-11e8-9644-59621ad5c975.PNG)

**Comment by molobrakos** [comment]
@fluppie I understand it now works for you as wellness?

**Comment by fluppie** [comment]
Yes indeed. I was pulling from your master branch and had to switch to the voc branch to make it work.
 I only propose a forced update/refresh of the plugin 10 seconds after pressing the start/stop button.

**Comment by molobrakos** [comment]
> Yes indeed. I was pulling from your master branch and had to switch to the voc branch to make it work.

Great!

> I only propose a forced update/refresh of the plugin 10 seconds after pressing the start/stop button.

Maybe you can try this yourself and add a call to `await asyncio.sleep(10)` between the call to `await self.call(...)` and await `self.update()` here: https://github.com/molobrakos/volvooncall/blob/master/volvooncall.py#L389

**Comment by molobrakos** [comment]
Closing this as fixed for now. Let me know how it works out with adding the delay.
